addons/mail/models/mail_thread.py

```diff
@@ -1625,7 +1630,9 @@ def message_parse(self, message, save_original=False):
         parent_message = self._get_parent_message(msg_dict)
         if parent_message:
             msg_dict['parent_id'] = parent_message.id
-            msg_dict['is_internal'] = bool(parent_message.subtype_id and parent_message.subtype_id.internal)
+            # If responding to a bot message, make comment as nobody would be notified otherwise
+            parent_sent_by_bot = parent_message.author_id == self.env['res.users'].browse(SUPERUSER_ID).partner_id
+            msg_dict['is_internal'] = bool(parent_message.subtype_id and parent_message.subtype_id.internal and not parent_sent_by_bot)
```

Comment:
:thinking: I know I'm the one who proposed the heuristic but the more I look at it the less safe it looks...

Maybe we should just change all the automated templates to explicitly post discussions? It's kind of a hack for a hack right now I guess since they're notifications to avoid notifying followers

At the risk of postponing this a bit more

Reply:
Yeah, actually we don't want to send automated messages in project to all followers (as you may have template in each stage). But we want to receive the answers ... 

Don't remember all discussion, but we could use a specific field for that instead of hacking the answer mechanism, indeed. But not sure what to do...

If we define a new subtype: yet another subtype people would have to manage. There are already several default subtypes (note, activity, comment). Moreover subtypes are used for finding recipients and display tweaks, not really to define the global management of the message.
Have a specific field on message for that ('reply_do_notify'), but not a fan of adding a column for that purpose. It will be ignored by most messages.
Have a new message_type. Currently we have comment, notification, email, sms, and it is used to characterize the source (I also think we should add mailing, for message created using the mailing mode; currently email is used for both outgoing and incoming messages, would like to split those two). We could have a new message_type ('comment_automated' ? 'comment_autoreply' ? ...) that would have its own specific behavior in mailgateway. As message has a specific source, answers to it could lead to specific behavior.

All in all, if we want to avoid the hack, would go for a specific message_type I think, seems less invasive and more coherent with current model of message.